content/en/docs/setup/install/virtual-machine/index.md

```diff
@@ -80,64 +85,36 @@ Install Istio and expose the control plane so that your virtual machine can acce
+    {{< /tip >}}
 
-    {{< text bash >}}
-    $ ISTIO_SERVICE_CIDR=$(echo '{"apiVersion":"v1","kind":"Service","metadata":{"name":"tst"},"spec":{"clusterIP":"1.1.1.1","ports":[{"port":443}]}}' | kubectl apply -f - 2>&1 | sed 's/.*valid IPs is //')
-    $ touch "${WORK_DIR}"/cluster.env
-    $ echo ISTIO_SERVICE_CIDR=$ISTIO_SERVICE_CIDR > "${WORK_DIR}"/cluster.env
-    {{< /text >}}
+1. Use the `istioctl x workload entry` command to generate:
+     * `cluster.env`: Contains metadata that identifies what namespace, service account, network CIDR and (optionally) what inbound ports to capture.
+     * `istio-token`: A Kubernetes token used to get certs from the CA.
+     * `mesh.yaml`: Provides additional Istio metadata including, network name, trust domain and other values.
+     * `root-cert.pem`: The root certificate used to authenticate.
+     * `hosts`: An addendum to `/etc/hosts` that the proxy will use to reach istiod for xDS.*
 
-1. Optionally configure a select set of ports for exposure from the
-   virtual machine. If you do not apply this optional step, all outbound traffic
-   on all ports is sent to the Kubernetes cluster. You may wish to send some
-   traffic on specific ports to other destinations. This example shows enabling
-   ports `3306` and `8080` for capture by Istio virtual machine integration and
-   transmission to Kubernetes. All other ports are sent over the default gateway
-   of the virtual machine.
+     {{< idea >}}
+     \*A sophisticated option involves configuring DNS within the virtual
+     machine to reference an external DNS server. This option is beyond
+     the scope of this guide.
+     {{< /idea >}}
 
     {{< text bash >}}
-    $ echo "ISTIO_INBOUND_PORTS=3306,8080" >> "${WORK_DIR}"/cluster.env
+    $ istioctl x workload entry configure -f workloadgroup.yaml -o "${WORK_DIR}"
     {{< /text >}}
 
-1. Add an IP address that represents Istiod. Replace `${INGRESS_HOST}` with the
-    ingress gateway service of istiod. Revisit
-    [Determining the ingress host and ports](/docs/tasks/traffic-management/ingress/ingress-control/#determining-the-ingress-ip-and-ports) to set the environment variable `${INGRESS_HOST}`.
-
-    {{< text bash >}}
-    $ touch "${WORK_DIR}"/hosts-addendum
-    $ echo "${INGRESS_HOST} istiod.istio-system.svc" > "${WORK_DIR}"/hosts-addendum
-    {{< /text >}}
-
-    {{< idea >}}
-    A sophisticated option involves configuring DNS within the virtual
-    machine to reference an external DNS server. This option is beyond
-    the scope of this guide.
-    {{< /idea >}}
-
-1. Create `sidecar.env` file to import the required environment variables:
-
-    {{< text bash >}}
-    $ touch "${WORK_DIR}"/sidecar.env
-    $ echo "PROV_CERT=/var/run/secrets/istio" >>"${WORK_DIR}"/sidecar.env
-    $ echo "OUTPUT_CERTS=/var/run/secrets/istio" >> "${WORK_DIR}"/sidecar.env
-    $ echo "ISTIO_NAMESPACE=${VM_NAMESPACE}" >> "${WORK_DIR}"/sidecar.env
-    {{< /text >}}
+    {{< tip >}}
```

Comment:
just a structural thing, we should probably have two paths, one with auto registration and one without, rather than interspersing them. Maybe with tabs?